File: ancillaryscreens/src/main/java/com/samagra/ancillaryscreens/data/model/AssessmentService.java
package com.samagra.ancillaryscreens.data.model;

import com.google.gson.JsonElement;
import com.morziz.network.annotation.Retry;
import com.morziz.network.annotation.RetryPolicyType;
import com.samagra.ancillaryscreens.data.login.LoginModel;
import com.samagra.ancillaryscreens.data.otp.ApiResponseModel;
import com.samagra.ancillaryscreens.data.otp.LoginRequest;
import com.samagra.ancillaryscreens.data.otp.Status;
import com.samagra.commons.models.submitresultsdata.SubmitResultsModel;
import com.samagra.commons.models.surveydata.SurveyModel;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface AssessmentService {
    @POST("sendOTP")
    Observable<ApiResponseModel<Status>> sendOtp(@Query("phone") String phoneNo,
                                                 @Query("errorMessage") String errorMessage);

    @POST("verifyOTP")
    Observable<LoginModel> verifyOtp(
            @Query("phone") String phoneNo,
            @Query("otp") String otp
    );

    @POST("assessment-visit-results")
    @Retry(retryPolicy = RetryPolicyType.linear, retryCount = 3)
    Call<Response<JsonElement>> insertVisitsResultsSync(@Header("authorization") String apiKey,
                                                        @Body List<SubmitResultsModel> finalResult);

    @POST("assessment-survey-results")
    Call<Response<JsonElement>> insertSurvey(@Header("authorization") String apiKey,
                                             @Body List<SurveyModel> survey);

    @POST("/api/school/{udise}/result/calculate")
    Call<Void> postSchoolSubmission(@Header("authorization") String apiKey, @Path("udise") Long udise, @Query("cycle_id") int cycleId);
}

File: data/src/main/java/com/data/network/AssessmentService.kt
package com.data.network

import com.data.db.models.StudentAssessmentHistoryCompleteInfo
import com.data.models.history.GradeSummary
import com.data.models.submissions.SubmitResultsModel
import com.morziz.network.annotation.Retry
import com.morziz.network.annotation.RetryPolicyType
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface AssessmentService {
    @GET("school/{udise}/students/result/summary")
    @Retry(retryPolicy = RetryPolicyType.exponential, retryCount = 2)
    fun getAssessmentHistories(
        @Path("udise") udise: String,
        @Query("grade") grades : String,
        @Header("Accept-Language") lang: String,
        @Header("authorization") token: String
    ): Call<MutableList<GradeSummary>>


    @GET("school/{udise}/students/result")
    @Retry(retryPolicy = RetryPolicyType.exponential, retryCount = 2)
    fun getSchoolStatusHistories(
        @Path("udise") udise: String,
        @Query("grade") grades : String,
        @Query("cycle_id") cycleId : Int,
        @Header("Accept-Language") lang: String,
        @Header("authorization") token: String
    ): Call<MutableList<StudentAssessmentHistoryCompleteInfo>>

    @POST("assessment-visit-results")
    fun postSubmissions(
        @Header("authorization") apiKey: String,
        @Body finalResult: List<SubmitResultsModel>
    ): Call<Void>

    @POST("assessment-visit-results")
    fun postSubmissionsForDeletedStudents(
        @Header("authorization") apiKey: String,
        @Query("trigger") trigger: String,
        @Body finalResult: List<SubmitResultsModel>
    ): Call<Void>

}